compiler/z/codegen/S390GenerateInstructions.cpp

```diff
@@ -2131,23 +2131,33 @@ TR::Instruction * generateVSIInstruction(
    }
 
 /************************************************************ Misc Instructions ************************************************************/
-TR::Instruction *
-generateS390PseudoInstruction(TR::CodeGenerator * cg, TR::InstOpCode::Mnemonic op, TR::Node * n, TR::Node * fenceNode, TR::Instruction * preced)
+TR::Instruction*
+generateS390PseudoInstruction(TR::CodeGenerator* cg, TR::InstOpCode::Mnemonic op, TR::Node* n, TR::Node* fenceNode, TR::Instruction* preced)
    {
-   TR::S390PseudoInstruction *instr;
-   TR::Compilation *comp = cg->comp();
+   TR::Instruction* cursor;
 
    if (preced)
-   {
-     instr= new (INSN_HEAP) TR::S390PseudoInstruction(op, n, fenceNode, preced, cg);
-   }
+      {
+      cursor = new (INSN_HEAP) TR::S390PseudoInstruction(op, n, fenceNode, preced, cg);
+      }
    else
-   {
-     instr= new (INSN_HEAP) TR::S390PseudoInstruction(op, n, fenceNode, cg);
-   }
+      {
+      cursor = new (INSN_HEAP) TR::S390PseudoInstruction(op, n, fenceNode, cg);
+      }
 
+   switch (op)
+      {
+      case TR::InstOpCode::PROC:
+         {
+         if (cg->comp()->getOption(TR_EntryBreakPoints))
+            {
+            cursor = new (INSN_HEAP) TR::S390EInstruction(TR::InstOpCode::BREAK, n, cursor, cg);
```

Comment:
Apologies for not being 100% familiar with the order in which the Z codegen emits things.  Will this breakpoint appear after the prologue then?  Is that the behaviour you want?

Reply:
Yes it should appear after the prologue. You're right it is probably best if the break got inserted before. I'll swap the order of generation.

Reply:
Fixed in 1359fee